### PR TITLE
Switch CI to use roles

### PR DESCRIPTION
**Pull request by DmitriyMusatkin**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -124,7 +158,11 @@ jobs:
   linux-aws-lc-fips:
     runs-on: ubuntu-22.04 # latest
```
Comment (graebm) [inline .github/workflows/ci.yml:159]: you changed all the other ubuntu to be 24.04, but this one is still 22.04 Did you miss this one by accident? If you left it 22.04 on purpose, leave a comment why

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -48,45 +54,65 @@ jobs:
...
+    - name: Build ${{ env.PACKAGE_NAME }}
+      run: |
+        aws s3 cp s3://aws-crt-test-stuff/ci/${{ env.BUILDER_VERSION }}/linux-container-ci.sh ./linux-container-ci.sh && chmod a+x ./linux-container-ci.sh
+        ./linux-container-ci.sh ${{ env.BUILDER_VERSION }} aws-crt-${{ env.LINUX_BASE_IMAGE }} build -p ${{ env.PACKAGE_NAME }} --cmake-extra=-DBUILD_SHARED_LIBS=ON
 
   #Warning: this ci relies on ubuntu pulling correct version of openssl
   #Current version (18.04) pulls 1.1.1, but 22.04+ will pull in 3.0
   linux-openssl-static:
-    runs-on: ubuntu-20.04 # latest
+    runs-on: ubuntu-24.04 # latest
```
Comment (graebm) [inline .github/workflows/ci.yml:109]: This comment refers to Ubuntu 18 and 22. Update it to reflect new reality.

Just delete comment if there's nothing interesting going on when using 24.04

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -95,25 +121,33 @@ jobs:
   #ubuntu 22.04 defaults to openssl3 version by default when installing
   #libssl-dev package. Hence we can rely on OS version to pull in openssl3.
   linux-openssl3-static:
-    runs-on: ubuntu-22.04 # latest
+    runs-on: ubuntu-24.04 # latest
```
Comment (graebm) [inline .github/workflows/ci.yml:124]: same thing about comment

File: .github/workflows/ci.yml (hunk 2 above)
Comment (DmitriyMusatkin) [reply .github/workflows/ci.yml:109]: it refers to base image in this case, which is still 18

File: .github/workflows/ci.yml (hunk 3 above)
Comment (DmitriyMusatkin) [reply .github/workflows/ci.yml:124]: same thing here. this is just updating what it runs on

**Pull request merged**